### PR TITLE
Add code approval support

### PR DESCRIPTION
**Pull request by davidx**
This enables support for github code approvals.
When thumbs counts code reviews to compare to minimum_reviewers, it will now count the code approvals also.
You may use comment based +1 as well as github code approvals interchangeably.(it adds up both) (it doesnt count twice if a user does a +1 And a code review also) 

This was implemented using the new github graphql api as code approvals are only available via the events webhook service, and graphql, not yet on the normal rest api.
Likely to be refactored once github releases more support for the feature in their api.

Fixes:
- It includes a fix for code review time horizon to consider base time stamp when it is newer. 
problem example: build shows 1/2 after being triggered by a base merge https://github.com/basho/riak_ql/pull/141
- Slashes in ref name

Features:
- retry command

### CODE REVIEW

File: lib/thumbs.rb [hunk 1]
```diff
@@ -7,13 +7,51 @@
...
+  module GitHub
+    HTTP = GraphQL::Client::HTTP.new("https://api.github.com/graphql") do
+      def headers(context)
+        {
+            "Authorization" => "Bearer #{ENV['GITHUB_ACCESS_TOKEN']}"
+        }
+      end
+    end
+    file='schema.json'
+    dump(file) unless File.exists?(file)
```
Comment (paegun) [inline lib/thumbs.rb:44]: Assumedly the `schema.json` content may change in the future, so this unless is blocking automatic updating. That said, if we know a schema change is available and required, something very likely will break downstream, so I would not argue to change this.

**Comment by paegun** [comment]
+1

File: lib/thumbs.rb (hunk 1 above)
Comment (davidx) [reply lib/thumbs.rb:44]: Yeh we may want to just check in the schema.json, while also perhaps improving upon this import code. for now it'll just use the one it loaded first. Thanks!

**Comment by davidx** [comment]
tests failed because multiple prs were building using nearly similar test suite that is not completely isolated, so it was creating and deleting tmp directories that both builds needed.

**Comment by davidx** [comment]
@javajolt, @paegun, @ooshlablu  ^^   😄  good to merge? 😄

**Comment by paegun** [comment]
+1

**Comment by davidx** [comment]
Thanks! 1/2

**Comment by hazen** [comment]
+1

**Comment by davidx** [comment]
Thanks!!

**Pull request merged**